docs/features/software-templates/writing-custom-actions.md

```diff
@@ -141,6 +141,90 @@ Prefer to use `camelCase` over `snake_case` or `kebab-case` for these actions if
 
 > We're aware that there are some exceptions to this, but try to follow as close as possible. We'll be working on migrating these in the repository over time too.
 
+### Adding a TemplateExample
+
+A TemplateExample is a predefined structure that can be used to create custom actions in your software templates. It serves as a blueprint for users to understand how to use a specific action and its fields as well as to ensure consistency and standardization across different custom actions.
+
+#### How to Use a TemplateExample
+
+1. Identify the custom action you want to create.
+2. Refer to the TemplateExample to understand the required structure and components.
+3. Fill in the necessary details in the template, such as action name, parameters, and logic.
+4. Integrate the completed template into your software template to enable the custom action.
+
+#### Benefits
+
+- Serves as templates for users to understand how to configure and use the actions.
+- Ensures consistency in custom actions.
+- Simplifies the creation process by providing a clear structure.
+- Helps in maintaining standardization across different templates.
```

Comment:
```suggestion
```

I think your initial text is great and explains why this is a great thing to have, let's keep it short and sweet :)